src/devices/Ssd1351/Ssd1351.cs

```diff
@@ -0,0 +1,295 @@
+            {
+                (colourBytes[0], colourBytes[1]) = Color565(color);
+            }
+            else
+            {
+                colourBytes[0] = (byte)((_colorSequence == ColorSequence.ABC ? color.B : color.R) >> 2);
+                colourBytes[1] = (byte)(color.G >> 2);
+                colourBytes[2] = (byte)((_colorSequence == ColorSequence.ABC ? color.R : color.B) >> 2);
+            }
+
+            
+            // set the pixels in the array representing the raw data to be sent to the display
+            // to the fill color
+            for (int i = 0; i < w * h; i++)
+            {
+                if (_colorDepth == ColorDepth.ColourDepth65K)
+                {
+                    displayBytes[i * 2 + 0] = colourBytes[0];
+                    displayBytes[i * 2 + 1] = colourBytes[1];
+                }
+                else
+                {
+                    displayBytes[i * 3 + 0] = colourBytes[0];
+                    displayBytes[i * 3 + 1] = colourBytes[1];
+                    displayBytes[i * 3 + 2] = colourBytes[2];
+                }
+            }
+
+            // specifiy a location for the rows and columns on the display where the data is to be written
+            SetColumnAddress(x, (byte)(x + w - 1));
+            SetRowAddress(y, (byte)(y + h - 1));
+
+            // write out the pixel data
+            SendCommand(Ssd1351Command.WriteRam, displayBytes);
+
+        }
+
+        public void ClearScreen()
+        {
+            FillRect(Color.Black, 0, 0, SCREEN_HEIGHT_PX, SCREEN_WIDTH_PX);
+        }
+
+        /// <summary>
+        /// Resets the display.
+        /// </summary>
+        public async Task ResetDisplayAsync()
+        {
+            _gpioDevice.Write(_dcPinId, PinValue.Low);
+            _gpioDevice.Write(_resetPinId, PinValue.High);
+
+            await Task.Delay(20);
+
+            _gpioDevice.Write(_resetPinId, PinValue.Low);
+            await Task.Delay(20);
+
+            _gpioDevice.Write(_resetPinId, PinValue.High);
+            await Task.Delay(20);
+        }
+
+        /// <summary>
+        /// Verifies value is within a specific range.
+        /// </summary>
+        /// <param name="value">Value to check.</param>
+        /// <param name="start">Starting value of range.</param>
+        /// <param name="end">Ending value of range.</param>
+        /// <returns>Determines if value is within range.</returns>
+        internal static bool InRange(uint value, uint start, uint end)
+        {
+            return (value - start) <= (end - start);
+        }
+
+        /// <summary>
+        /// Send a command to the the display controller along with associated parameters.
+        /// </summary>
+        /// <param name="command">Command to send.</param>
+        /// <param name="commandParameters">parameteters for the command to be sent</param>
+        public void SendCommand(Ssd1351Command command, params byte[] commandParameters)
```

Comment:
Should SendCommand methods be public? Is there a lot that you can do there that you can't do with the wrappers that you provided in the Commands file?

Reply:
I hadn't done all the commands but now I have. and so the methods are now private in the latest update.